### ISSUE TITLE
single-transition should use shorthand

### ISSUE THREAD

**Issue by paulirish** [Compass/compass#585]
``` css
  @include single-transition(all, 0.6s, ease-out);
```

will render the following:

``` css
  -moz-transition-property: all;
  -webkit-transition-property: all;
  -o-transition-property: all;
  transition-property: all;
  -moz-transition-duration: 0.6s;
  -webkit-transition-duration: 0.6s;
  -o-transition-duration: 0.6s;
  transition-duration: 0.6s;
  -moz-transition-timing-function: ease-out;
  -webkit-transition-timing-function: ease-out;
  -o-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
```

however.. it should render something much more clean like:

``` css
  -webkit-transition: all 0.6s ease-out;  
     -moz-transition: all 0.6s ease-out;  
       -o-transition: all 0.6s ease-out;  
          transition: all 0.6s ease-out;  
```

(without the fancy indentation if you wanna be like that.. :)

**Comment by davidchambers** [comment]
From [CSS Transitions Module](http://www.w3.org/TR/css3-transitions/#the-transition-shorthand-property-):

> The ‘[transition](http://www.w3.org/TR/css3-transitions/#transition)’ shorthand property combines the four properties described above into a single property.

It appears that when [WebKit added `-webkit-transition`](http://www.webkit.org/blog/138/css-animation/), the delay property did not exist. What would be the desired output for the following?

```
@include single-transition(all, 0.6s, ease-out, 0.2s);
```

What happens, for example, if a browser that doesn't know about the delay property is given `-webkit-transition: all 0.6s ease-out 0.2s`? Will it ignore the delay and apply the other values, or will it ignore the rule entirely?

**Comment by paulirish** [comment]
it would likely ignore the entire rule. 

I wonder what live webkit browsers this affects. My gut tells me we're definitely past them on desktop and on mobile they are dated. But that'd require some research.

In the meantime we can do:
- shorthand for everyone if delay prop is **not** provided
- shorthand for everyone but webkit if delay prop is provided
  - webkit gets shorthand minus delay and then longhand delay.

**Comment by davidchambers** [comment]
Sounds good. I'm happy to work on a patch for this; I'm keen to become more familiar with Compass's internals.

**Comment by paulirish** [comment]
closed by #593 it appears
